ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/SingleParticipantTestContext.scala

```diff
@@ -663,6 +667,37 @@ final class SingleParticipantTestContext private[participant] (
     services.command
       .submitAndWaitForTransactionTree(request)
 
+  /** This addresses a narrow case in which we tolerate a
+    * single occurrence of a specific and transient (and rare) error
+    * by retrying only a single time.
+    */
+  override def submitRequestAndTolerateGrpcError[T](
+      errorCodeToTolerateOnce: ErrorCode,
+      submitAndWaitGeneric: ParticipantTestContext => Future[T],
+  ): Future[T] = for {
+    // Issue the `submitAndWaitGeneric` submit the first (and for the
+    // most part the only) time. Returns a top-level Left if we
+    // encounter the error that we are supposed to tolerate.
+    eitherInFlightOrSuccess <- submitAndWaitGeneric(this).transform {
+      case Failure(e: StatusRuntimeException)
+          if errorCodeToTolerateOnce.category.grpcCode
+            .map(_.value())
+            .contains(StatusProto.fromThrowable(e).getCode) =>
+        Success(Left(e))
+      case otherTry =>
+        // Otherwise return a Right with a nested Either that
+        // let's us create a failed or successful future in the
+        // default case of the step below.
+        Success(Right(otherTry.toEither))
+    }
+    originalOrRetry <- eitherInFlightOrSuccess.fold(
+      _ =>
+        // If we are retrying a single time, back off first for one second.
+        Delayed.Future.by(1.second)(submitAndWaitGeneric(this)),
```

Comment:
Should the delay be configurable by the client? Is 1.second enough for the canton race condition in this case?

Reply:
I was actually more concerned that it may be too much, but erring on the conservative here I think. In the flakes we were looking at tens of milliseconds, but I figured wait as long as 1 second wouldn't hurt.